src/app/public/modules/split-view/split-view.component.ts

```diff
@@ -0,0 +1,375 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output
+} from '@angular/core';
+
+import {
+  animate,
+  style,
+  transition,
+  trigger
+} from '@angular/animations';
+
+import {
+  SkyAppWindowRef,
+  SkyCoreAdapterService,
+  SkyMediaBreakpoints,
+  SkyMediaQueryService
+} from '@skyux/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import {
+  Subscription
+} from 'rxjs/Subscription';
+
+import 'rxjs/add/operator/take';
+
+import 'rxjs/add/operator/takeUntil';
+
+import 'rxjs/add/operator/takeWhile';
+
+import 'rxjs/add/observable/fromEvent';
+
+import {
+  SkySplitViewBeforeWorkspaceCloseHandler
+} from './types/split-view-before-workspace-close-handler';
+
+import {
+  SkySplitViewMessage
+} from './types/split-view-message';
+
+import {
+  SkySplitViewMessageType
+} from './types/split-view-message-type';
+
+import {
+  SkySplitViewMediaQueryService
+} from './split-view-media-query.service';
+
+import {
+  SkySplitViewDrawerComponent
+} from './split-view-drawer.component';
+
+import {
+  SkySplitViewWorkspaceComponent
+} from './split-view-workspace.component';
+
+let nextId = 0;
+
+@Component({
+  selector: 'sky-split-view',
+  templateUrl: './split-view.component.html',
+  styleUrls: ['./split-view.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger(
+      'blockAnimationOnLoad', [
+        transition(':enter', [])
+      ]
+    ),
+    trigger(
+      'drawerEnter', [
+        transition(`void => *`, [
+          style({transform: 'translate(-100%)'}),
+          animate('150ms ease-in')
+        ])
+      ]
+    ),
+    trigger(
+      'workspaceEnter', [
+        transition(`void => *`, [
+          style({transform: 'translate(100%)'}),
+          animate('150ms ease-in')
+        ])
+      ]
+    )
+  ]
+})
+export class SkySplitViewComponent implements OnInit, AfterViewInit, OnDestroy {
+
+  @Input()
+  public backLabel: string;
+
+  @Input()
+  public set drawerWidth(value: number) {
```

Comment:
Remind me, we're going to try and put this input on the `drawer` component? If so, what do you think about naming it `defaultWidth` or `preferredWidth`?

Reply:
Yep, working on that. I've just got it named `width` for now, because a consumer could change it whenever they want and the template will react. The words `default` or `preferred` make me think there's something more to the intent, when all it really is doing is setting width. Would you be cool with just `width` ?

Reply:
The reason I requested a "looser" modifier is because the width can be changed by the user with the resize handle. So, the developer can set a "preferred width", but the user can override that width at will. Using a blatant "width" communicates that the width of the drawer is static. Is that accurate?

Reply:
All child-specific properties have been moved to the children components (width, ariaLabel). 👍